base/bin/build.sh

```diff
@@ -1,27 +1,188 @@
 
-# Change the current working directory to the `base` directory instead of
-# the `base/bin` directory
-pushd `dirname $0`/.. > /dev/null
+# fail-fast
+set -e
 
-RUNTIME_NAME="nodejs"
-DOCKER_NAMESPACE=${1}
-CANDIDATE_NAME=${2}
+function fail() {
+  echo "ERROR: ${1}"
+  exit 1
+}
+
+function print() {
+  if [ "${VERBOSE}" = "true" ]; then
+    echo "${1}"
+  fi
+}
+
+_NODE_VERSION_OPTION="--node-version"
+_UPLOAD_TO_STAGING_OPTION="--upload-to-staging"
+_CANDIDATE_NAME_OPTION="--candidate-name"
+_DOCKER_NAMESPACE_OPTION="--docker-namespace"
+_VERBOSE_OPTION="--verbose"
+_HELP_OPTION="--help"
+
+_DEFAULT="default"
 
-if [ -z "${DOCKER_NAMESPACE}" -o -z "${CANDIDATE_NAME}" ]; then
-  echo "Usage: ${0} <docker namespace> <candidate name>"
-  echo "Please provide release a docker namespace and candidate name."
+function usage() {
+  echo "Usage: $(basename $0) [options...]"
+  echo ""
+  echo "Build the Node.js runtime image"
+  echo ""
+  echo "Options:"
+  echo "  ${_DOCKER_NAMESPACE_OPTION}   Specifies the namespace to use for the build image, "
+  echo "                       (For example \"gcr.io/my-project\")."
+  echo ""
+  echo "                       If not specified, or if the value \"${_DEFAULT}\" is specified, then"
+  echo "                       \"gcr.io/<current gcloud project>\" will be used."
+  echo "  ${_CANDIDATE_NAME_OPTION}     Specifies a candiate name to use for the image.  This will be"
+  echo "                       used to tag the image."
+  echo ""
+  echo "                       If unspecified, or if the value \"${_DEFAULT}\" is specified, the current"
+  echo "                       timestamp will be used."
+  echo "  ${_NODE_VERSION_OPTION}       Specify the Node.js version in the runtime image that will be"
+  echo "                       used as a tag for the built image."
+  echo ""
+  echo "                       If not specified, or if the value \"${_DEFAULT}\" is specified, the"
+  echo "                       built image will not be tagged with a version."
+  echo "  ${_UPLOAD_TO_STAGING_OPTION}  Specifies whether to mark the built image with the 'staging' tag."
+  echo "                       Must be followed by either \"true\", \"false\" or \"${_DEFAULT}\"."
+  echo ""
+  echo "                       If not specified or if the value \"${_DEFAULT}\" is specified, the image "
+  echo "                       will not be given the 'staging' tag."
+  echo "  ${_VERBOSE_OPTION}            Causes details to be printed as this command is run"
+  echo "  ${_HELP_OPTION}               Displays this help"
   exit 1
+}
+
+DOCKER_NAMESPACE="${_DEFAULT}"
+CANDIDATE_NAME="${_DEFAULT}"
+NODE_VERSION_TAG="${_DEFAULT}"
+UPLOAD_TO_STAGING="${_DEFAULT}"
+
+VERBOSE="false"
+RUNTIME_NAME="nodejs"
+
+# Parse the command line arguments
+while [ $# -gt 0 ]; do
+  case "$1" in
+    ${_NODE_VERSION_OPTION})
+      shift
+      if [ -z "${1}" ]; then
+        fail "${_NODE_VERSION_OPTION} must be followed by a version string"
```

Comment:
Do you need to safe guard against input like `--docker-namespace --upload-to-staging`?